specs/vep-1-vdk-apache-airflow-provider/README.MD

```diff
@@ -0,0 +1,356 @@
+
+Providers can contain operators, sensors, hooks, and transfer operators to communicate
+with a multitude of external systems, but they can also extend Airflow core with new
+capabilities.
+
+The proposed design describes the solution for supporting Multistage Data Pipelines by
+integrating VDK with Apache Airflow. For this purpose, we!!! will implement an Apache
+Airflow Provider called "VDK Provider" that will be responsible for managing Data Jobs.
+It will give us the ability to define DAGs in which the tasks (jobs) will be defined by
+instantiating custom operators. The operators will start and monitor Data Job executions
+via the Control Service Execution API.
+
+In order to speed up the VDK popularization and provide easier integration with Airflow
+to our customers, the VDK Provider will be contributed to the Airflow Community.
+
+## API design
+
+No changes to the public API.
+
+## Detailed design
+
+### VDK Provider
+
+The integration of VDK with Apache Airflow will be accomplished by VDK Provider
+implementation. It will provide an ability to the customers to define Airflow Workflow
+that consists of Data Jobs. Basically, it will be responsible for managing Data Job
+executions - starting, canceling, retrieving. As can be seen from the below diagram the
+provider will consist of a hook, operator, and sensor.
+
+![img_7.png](img_7.png)
+
+Customers will be able to choose from two operating modes - synchronous and asynchronous,
+depending on their Data Jobs and needs. The synchronous mode will be achieved by
+providing an operator derived from `airflow.models.baseoperator.BaseOperator`, which
+executes a Data Job and waits for its completion. The asynchronous mode will be achieved
+by providing a sensor operator derived from `airflow.sensors.base.BaseSensorOperator`,
+which polls on the specific time intervals for job execution completion and does not hold
+the Airflow Worker.
+
+<!--
+* Capacity Estimation and Constraints
+    * Cost of data path: CPU cost per-IO, memory footprint, network footprint.
+    * Cost of control plane including cost of APIs, expected timeliness from layers
+      above.
+* Availability.
+    * For example - is it tolerant to failures, What happens when the service stops
+      working
+* Performance.
+    * Consider performance of data operations for different types of workloads. Consider
+      performance of control operations
+    * Consider performance under steady state as well under various pathological
+      scenarios, e.g., different failure cases, partitioning, recovery.
+    * Performance scalability along different dimensions, e.g. #objects, network
+      properties (latency, bandwidth), number of data jobs, processed/ingested data, etc.
+* Telemetry and monitoring changes (new metrics).
+* Configuration changes.
+* Upgrade / Downgrade Strategy (especially if it might be breaking change).
+* Troubleshooting
+    * What are possible failure modes.
```

Comment:
This definitely deserves some answer. It's completely fine if it's for another PR. 

Now that users would schedule jobs using airflow.What happens if Job x fails. What if it fails with user error or platform error. How are restarts handled (by airflow or by vdk runtime).